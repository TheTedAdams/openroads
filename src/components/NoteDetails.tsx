import * as React from 'react';
import { INotesDetailsProps } from "../types/types"

class NoteDetails extends React.Component<INotesDetailsProps> {
  render() {
    const { notes, currentIndex, markAsRead } = this.props
    return(
      <section className='NoteDetails'>
        {/* fixed: rendering bugs when list is empty */}
        {notes.length > 0 && (
          <>
            <h3 className='NoteDetails-title'>{notes[currentIndex].subject}</h3>
            <p>{notes[currentIndex].body}</p>
            <button disabled={notes[currentIndex].read === true} onClick={markAsRead}>Mark as read</button>
          </>
        )}
      </section>
    );
  }
}

export default NoteDetails