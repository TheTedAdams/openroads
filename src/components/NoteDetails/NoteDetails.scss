.NoteDetails {
  grid-column: 2;
  grid-row: 2;
  padding: 1rem;
  overflow-y: auto;
}
