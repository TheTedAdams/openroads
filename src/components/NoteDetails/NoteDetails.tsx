import './NoteDetails.scss';

import React, { FC } from 'react';
import { Note } from '../../models';

interface NoteDetailsProps {
  isLoading: boolean;
  selectedNote: Note;
  onMarkAsRead: () => void;
}

export const NoteDetails: FC<NoteDetailsProps> = ({
  isLoading,
  selectedNote,
  onMarkAsRead,
}) => {
  return (
    <section className="NoteDetails">
      {isLoading ? (
        <p>Loading...</p>
      ) : !selectedNote ? (
        <p>No note selected.</p>
      ) : (
        <>
          <h3 className="NoteDetails-title">{selectedNote.subject}</h3>
          <p className="NoteDetails-subject">{selectedNote.body}</p>
          <button onClick={onMarkAsRead}>Mark as read</button>
        </>
      )}
    </section>
  );
};
