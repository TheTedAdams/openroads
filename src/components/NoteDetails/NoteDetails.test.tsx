import React from 'react';
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';

import { NoteDetails } from './NoteDetails';

const TEST_NOTE = { subject: 'SUBJECT', body: 'BODY', read: false };

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(
    <NoteDetails
      isLoading={false}
      onMarkAsRead={() => null}
      selectedNote={TEST_NOTE}
    />,
    div
  );
  ReactDOM.unmountComponentAtNode(div);
});

it('Renders like the snapshot', () => {
  const tree = renderer
    .create(
      <NoteDetails
        isLoading={false}
        onMarkAsRead={() => null}
        selectedNote={TEST_NOTE}
      />
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
