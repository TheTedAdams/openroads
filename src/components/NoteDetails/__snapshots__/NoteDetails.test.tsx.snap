// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders like the snapshot 1`] = `
<section
  className="NoteDetails"
>
  <h3
    className="NoteDetails-title"
  >
    SUBJECT
  </h3>
  <p
    className="NoteDetails-subject"
  >
    BODY
  </p>
  <button
    onClick={[Function]}
  >
    Mark as read
  </button>
</section>
`;
